src/log-store/src/kafka/client_manager.rs

```diff
@@ -134,3 +134,77 @@ impl ClientManager {
         Ok(Client::new(raw_client, &self.config))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use common_test_util::get_broker_endpoints;
+    use common_test_util::wal::kafka::topic_decorator::{Affix, TopicDecorator};
+    use common_test_util::wal::kafka::{create_topics, BROKER_ENDPOINTS_KEY};
+
+    use super::*;
+
+    /// Prepares for a test in that a collection of topics and a client manager are created.
+    async fn prepare(test_name: &str, num_topics: usize) -> (ClientManager, Vec<Topic>) {
+        let broker_endpoints = get_broker_endpoints!(BROKER_ENDPOINTS_KEY);
+        let decorator = TopicDecorator::default()
+            .with_prefix(Affix::Fixed(test_name.to_string()))
+            .with_suffix(Affix::TimeNow);
+        let topics = create_topics(num_topics, decorator, &broker_endpoints, None).await;
+
+        let config = KafkaConfig {
+            broker_endpoints,
+            ..Default::default()
+        };
+        let manager = ClientManager::try_new(&config).await.unwrap();
+
+        (manager, topics)
+    }
+
+    /// Checks clients for the given topics are created.
+    async fn ensure_clients_exist(topics: &[Topic], client_manager: &ClientManager) {
+        let client_pool = client_manager.client_pool.read().await;
+        let all_exist = topics.iter().all(|topic| client_pool.contains_key(topic));
+        assert!(all_exist);
+    }
+
+    async fn test_with(test_name: &str) {
+        let (manager, topics) = prepare(test_name, 128).await;
+        // Assigns multiple regions to a topic.
+        let region_topic = (0..512)
+            .map(|region_id| (region_id, &topics[region_id % topics.len()]))
+            .collect::<HashMap<_, _>>();
+
+        // Gets the assigned topic for each region and then gets the associated client.
+        match test_name {
+            "test_sequential" => {
```

Comment:
Prefer not to use string here. An simple enum will do.

Reply:
Replaced with an enum

Reply:
The original design is an anti-pattern. We only use the `test_with` in two places. Would you be able to separate it into two methods?

Reply:
Separated into two methods.